target/linux/ipq40xx/base-files/lib/upgrade/platform.sh

```diff
@@ -78,6 +78,16 @@ platform_do_upgrade() {
 	compex,wpj419)
 		nand_do_upgrade "$1"
 		;;
+	engenius,eap2200)
+		target_mtd="$(cat /proc/mtd | grep APPSBLENV | awk -F: {'print $1'})"
```

Comment:
Would this work as well?

    target_mtd=$(find_mtd_part APPSBLENV)
    part=$(strings "$target_mtd" | sed -n 's/^'"active_fw"'=//p')

Reply:
Thanks for comment, modified the method.